publisher/publisher.go

```diff
@@ -117,20 +169,32 @@ func (pub *Impl) SubmitToSingleCT(ctx context.Context, logID string, der []byte)
 
 	chain := append([]ct.ASN1Cert{der}, pub.issuerBundle...)
 
-	ctLog := pub.logIDToLog(logID)
-	if ctLog == nil {
-		err := fmt.Errorf("Failed to find log matching ID: %s", logID)
-		pub.log.AuditErr(err.Error())
-		return err
+	// Try to find a *Log from the cache for this logURL. If this is the first
+	// request to submit to this log then add it to the cache
+	var ctLog *Log
+	if ctLog = pub.ctLogsCache.GetLog(logPublicKey); ctLog == nil {
```

Comment:
Since AddLog already does this check for you (and does it under a lock), doing it here is redundant.